store/store.go

```diff
@@ -379,25 +388,66 @@ func (s *store) IsBanned(addr crypto.Address) bool {
 	return s.config.BannedAddrs[addr]
 }
 
+func (s *store) Prune(resultFunc func(pruned, skipped, pruningHeight uint32)) error {
+	cert := s.lastCertificate()
+
+	// at genesis block
+	if cert == nil {
+		return nil
+	}
+
+	retentionBlocks := uint32(s.config.RetentionDays * totalBlockInDay)
+
+	if cert.Height() < retentionBlocks {
+		return nil
+	}
+
+	pruningHeight := cert.Height() - retentionBlocks
+
+	// start pruning blocks and transactions
+	for i := pruningHeight; i >= 1; i-- {
+		deleted, err := s.pruneBlock(i)
+		if err != nil {
+			return err
+		}
+
+		if deleted {
```

Comment:
If `deleted` is false we we had an error and we returned the function and this case won't be executed IMO.

Reply:
This skipped removed or not found block.

Reply:
But when its false we have an error and we won't continue, how this block get executed at all?

Reply:
> But when its false we have an error and we won't continue, how this block get executed at all?

- false + nil = skipped non exists block
- true + nil = block pruned
- false + err = error and stop process